### ISSUE TITLE
Support for arguments following different syntax

### ISSUE THREAD

**Issue by kiambogo** [gilt/nova#13]
Currently one can supply additional parameters to the application through `deployment_arguments`, which get appended to the docker run command. The syntax that Nova uses is `key=value` as defined here: https://github.com/gilt/nova/blob/master/nova/core/deploy/deploy_stack.py#L193.

This works well enough for many arguments, such as defining a config file in Play `-Dconfig.resource=/path/to/file`. However, some arguments like the JVM memory ones do not follow this syntax, but rather are simply `-Xmx256m`, or in the case of Play 2.3's `mem` argument, `-mem 256`. Currently, I believe, there is no way to specify params with this kind of syntax. 

As Nova, with the help of Docker, aims to be a general deployment tool (one not strictly used for JVM-based applications), a solution shouldn't be designed which only caters to JVM arguments. I think it would be beneficial to provide another optional section in the Nova specification where the developer can pass a list of strings which get passed as-is to the launch command:

Option 1:

    deployment_options:
        - --log-driver: syslog
    deployment_arguments:
        - -Dconfig.resource: application.production.conf
    deployment_strings:
        - '-mem 7168'
        - '-Xmx71668m'

Option 2:

    deployment_options:
        - --log-driver: syslog
    deployment_arguments:
        - -Dconfig.resource: application.production.conf
        - '-mem 7168'
        - '-Xmx7168m'

which would result in a launch command 

    docker run -d --log-driver=syslog org/docker_image -Dconfig.resource=application.production.conf -mem 7168 -Xmx7168m

This would allow for more flexibility in the varied syntax of arguments that could be specified to the application on launch.

**Comment by fiadliel** [comment]
Agreed. This shouldn't be a key/value list. And `key=value` shouldn't be treated in any special way - these should just be strings; each entry is one argument (including any spaces).
